### PR TITLE
[6.4.x] BZ1263753 - Disable link to process instance details when activity no…

### PR DESCRIPTION
**Pull request by cristianonicolai**
…t present

New perspective exclusive to Drools that removes the new task button

### CODE REVIEW

File: jbpm-console-ng-human-tasks/jbpm-console-ng-human-tasks-client/src/main/java/org/jbpm/console/ng/ht/client/editors/taskslist/grid/AbstractTasksListGridPresenter.java [hunk 1]
```diff
@@ -0,0 +1,443 @@
...
+            }
+        };
+    }
+
+    public AbstractTasksListGridPresenter(DataSetTasksListGridPresenter.DataSetTaskListView view,
+                                         Caller<TaskLifeCycleService> taskOperationsService,
+                                         DataSetQueryHelper dataSetQueryHelper,
+                                         DataSetQueryHelper dataSetQueryHelperDomainSpecific,
+                                         User identity
+    ) {
```
Comment (manstis) [inline jbpm-console-ng-human-tasks/jbpm-console-ng-human-tasks-client/src/main/java/org/jbpm/console/ng/ht/client/editors/taskslist/grid/AbstractTasksListGridPresenter.java]: Looks like you need to run a code formatter over this.. Sorry.

File: jbpm-console-ng-human-tasks/jbpm-console-ng-human-tasks-client/src/main/java/org/jbpm/console/ng/ht/client/editors/taskslist/grid/DroolsTasksListGridPresenter.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+@WorkbenchScreen(identifier = DroolsTasksListGridPresenter.SCREEN_ID)
+public class DroolsTasksListGridPresenter extends AbstractTasksListGridPresenter {
+
+    public static final String SCREEN_ID = "Drools Tasks List";
+
+    public DroolsTasksListGridPresenter() {
+        super();
+    }
+
+    public DroolsTasksListGridPresenter(
```
Comment (manstis) [inline jbpm-console-ng-human-tasks/jbpm-console-ng-human-tasks-client/src/main/java/org/jbpm/console/ng/ht/client/editors/taskslist/grid/DroolsTasksListGridPresenter.java]: I feel really bad calling out code formatting issues.... but.. well....

File: jbpm-console-ng-human-tasks/jbpm-console-ng-human-tasks-client/src/main/java/org/jbpm/console/ng/ht/client/editors/taskslist/grid/dash/DataSetTasksListGridPresenter.java [hunk 3]
```diff
@@ -15,428 +15,48 @@
...
-            }
-        }, new ErrorCallback<Message>() {
-            @Override
-            public boolean error( Message message,
-                                  Throwable throwable ) {
-                errorPopup.showMessage( Constants.INSTANCE.UnexpectedError(throwable.getMessage())  );
-                return true;
-            }
-        } ).claim(taskId, userId, deploymentId );
+    public DataSetTasksListGridPresenter(
```
Comment (manstis) [inline jbpm-console-ng-human-tasks/jbpm-console-ng-human-tasks-client/src/main/java/org/jbpm/console/ng/ht/client/editors/taskslist/grid/dash/DataSetTasksListGridPresenter.java]: Ditto, re: code format...

**Comment by manstis** [comment]
@cristianonicolai The changes appear to largely be copy, split and paste so I've not looked in any real detail.. other than some code formatting matters, this PR looks OK to merge.. Would you mind running the code formatter over the classes and I'll happily merge.

**Comment by cristianonicolai** [comment]
@manstis no worries, my bad not checking that. Please review the updated PR.

File: jbpm-console-ng-human-tasks/jbpm-console-ng-human-tasks-client/src/main/java/org/jbpm/console/ng/ht/client/editors/taskprocesscontext/TaskProcessContextPresenter.java [hunk 4]
```diff
@@ -61,25 +69,30 @@
...
-            Caller<TaskQueryService> taskQueryService,
-            Caller<DataServiceEntryPoint> dataServices,
-            Event<ProcessInstancesWithDetailsRequestEvent> processInstanceSelected
+    public TaskProcessContextPresenter(TaskProcessContextView view,
+                                       PlaceManager placeManager,
+                                       Caller<TaskQueryService> taskQueryService,
+                                       Caller<DataServiceEntryPoint> dataServices,
+                                       Event<ProcessInstancesWithDetailsRequestEvent> processInstanceSelected,
+                                       ActivityManager activityManager
     ) {
```
Comment (manstis) [inline jbpm-console-ng-human-tasks/jbpm-console-ng-human-tasks-client/src/main/java/org/jbpm/console/ng/ht/client/editors/taskprocesscontext/TaskProcessContextPresenter.java]: Sorry.. you still have "hanging" parenthesis on your constructors :(

File: jbpm-console-ng-human-tasks/jbpm-console-ng-human-tasks-client/src/main/java/org/jbpm/console/ng/ht/client/editors/taskslist/grid/AbstractTasksListGridPresenter.java [hunk 5]
```diff
@@ -0,0 +1,442 @@
...
+            }
+        };
+    }
+
+    public AbstractTasksListGridPresenter(DataSetTasksListGridPresenter.DataSetTaskListView view,
+                                          Caller<TaskLifeCycleService> taskOperationsService,
+                                          DataSetQueryHelper dataSetQueryHelper,
+                                          DataSetQueryHelper dataSetQueryHelperDomainSpecific,
+                                          User identity
+    ) {
```
Comment (manstis) [inline jbpm-console-ng-human-tasks/jbpm-console-ng-human-tasks-client/src/main/java/org/jbpm/console/ng/ht/client/editors/taskslist/grid/AbstractTasksListGridPresenter.java]: Sorry.. you still have "hanging" parenthesis on your constructors :(

**Comment by manstis** [comment]
@cristianonicolai A couple of tweaks and then I'll merge..

**Comment by cristianonicolai** [comment]
@manstis hopefully solved now :)

**Comment by manstis** [comment]
+1 to merge.

**Pull request merged**